control-plane/agents/src/bin/core/controller/scheduling/resources/mod.rs

```diff
@@ -101,6 +102,33 @@ impl PoolItemLister {
             None => vec![],
         }
     }
+    /// Get a list of replicas wrapped as ChildItem, for resize.
+    pub(crate) async fn list_for_resize(registry: &Registry, spec: &VolumeSpec) -> Vec<ChildItem> {
+        let replicas = registry.specs().volume_replicas(&spec.uuid);
+        let state_replicas = registry.replicas().await;
+        let pool_wrappers = registry.pool_wrappers().await;
+
+        replicas
+            .iter()
+            .filter_map(|replica_spec| {
+                let replica_spec = replica_spec.lock().clone();
+                let replica_state = state_replicas
+                    .iter()
+                    .find(|state| state.uuid == replica_spec.uuid);
+
+                let pool_id = replica_spec.pool.pool_name().clone();
+                pool_wrappers
+                    .iter()
+                    .find(|p| p.id == pool_id)
```

Comment:
No need to clone

```suggestion
                let pool_id = replica_spec.pool.pool_name();
                pool_wrappers
                    .iter()
                    .find(|p| &p.id == pool_id)
```

Reply:
This should rather be `p.id == *pool_id` then ? Comparing values, not addresses/references.

Reply:
It's the same right? In rust references aren't the same as other languages, so it's not compare pointers for example